### PR TITLE
Remove jsp/jstl dependencies from terasoluna-gfw-web #436

### PR DESCRIPTION
**Pull request by btmoritamta**
Please review #436

### CODE REVIEW

File: terasoluna-gfw-parent/pom.xml [hunk 1]
```diff
@@ -311,6 +313,19 @@
...
+            <!-- == Start Tomcat == -->
+            <dependency>
+                <groupId>org.apache.tomcat</groupId>
+                <artifactId>tomcat-jsp-api</artifactId>
+                <version>${tomcat8.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.apache.tomcat</groupId>
+                <artifactId>tomcat-servlet-api</artifactId>
+                <version>${tomcat8.version}</version>
```
Comment (ikeyat) [inline terasoluna-gfw-parent/pom.xml]: - [ ] Why don't you use ``javax.servlet-api`` defined in Spring IO Platform. This does not need to define in ``<dependencyManagement/>``.

http://docs.spring.io/platform/docs/current/reference/htmlsingle/

File: terasoluna-gfw-parent/pom.xml [hunk 2]
```diff
@@ -311,6 +313,19 @@
             </dependency>
             <!-- == End Spring IO Platform == -->
 
+            <!-- == Start Tomcat == -->
+            <dependency>
+                <groupId>org.apache.tomcat</groupId>
+                <artifactId>tomcat-jsp-api</artifactId>
+                <version>${tomcat8.version}</version>
```
Comment (ikeyat) [inline terasoluna-gfw-parent/pom.xml]: - [ ] Why don't you use ``javax.servlet.jsp-api`` defined in Spring IO Platform. This does not need to define in ``<dependencyManagement/>``.

http://docs.spring.io/platform/docs/current/reference/htmlsingle/

**Comment by ikeyat** [comment]
- [ ] Please add ``web-jsp`` to ``mvn-build-all.sh`` and ``mvn-build-all.bat``.

File: terasoluna-gfw-web/pom.xml [hunk 3]
```diff
@@ -82,18 +82,17 @@
...
         <dependency>
-            <groupId>org.apache.tomcat</groupId>
-            <artifactId>tomcat-jsp-api</artifactId>
-            <scope>provided</scope>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet.jsp</groupId>
+            <artifactId>javax.servlet.jsp-api</artifactId>
```
Comment (ikeyat) [inline terasoluna-gfw-web/pom.xml]: - [ ] remove JSP dependency from terasoluna-gfw-web

File: terasoluna-gfw-web-jsp/pom.xml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+    <dependencies>
+        <!-- == Begin TERASOLUNA == -->
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>terasoluna-gfw-common</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] probably, `terasoluna-fw-common` is unnecessary.

File: terasoluna-gfw-web-jsp/pom.xml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            <groupId>${project.groupId}</groupId>
+            <artifactId>terasoluna-gfw-web</artifactId>
+        </dependency>
+        <!-- == End TERASOLUNA == -->
+
+        <!-- == Begin Spring == -->
+        <!-- Spring MVC -->
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-webmvc</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] probably, `spring-webmvc` is unnecessary.

File: terasoluna-gfw-web-jsp/pom.xml [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+            <scope>test</scope>
+        </dependency>
+        <!-- == End Spring == -->
+
+        <!-- == Begin Tomcat == -->
+        <!-- Servlet API 3.0 & JSP API 2.2 -->
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+        </dependency>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] please add `provided` scope

File: terasoluna-gfw-web-jsp/pom.xml [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        <dependency>
+            <groupId>javax.servlet.jsp</groupId>
+            <artifactId>javax.servlet.jsp-api</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] please add `provided` scope

File: terasoluna-gfw-web-jsp/pom.xml [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+            <groupId>org.apache.taglibs</groupId>
+            <artifactId>taglibs-standard-jstlel</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>javax.el</groupId>
+            <artifactId>javax.el-api</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] please add `provided` scope

File: terasoluna-gfw-web-jsp/pom.xml [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+            <artifactId>javax.servlet.jsp-api</artifactId>
+        </dependency>
+        <!-- == End Tomcat == -->
+
+        <!-- == Begin Other JavaEE/SE APIs == -->
+        <dependency>
+            <groupId>org.apache.taglibs</groupId>
+            <artifactId>taglibs-standard-jstlel</artifactId>
+        </dependency>
+        <dependency>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] probably, `javax.inject` is unnecessary.

File: terasoluna-gfw-web-jsp/pom.xml [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+            <groupId>javax.el</groupId>
+            <artifactId>javax.el-api</artifactId>
+        </dependency>
+        <!-- == End Other JavaEE/SE APIs == -->
+
+        <!-- == Begin Database == -->
+        <!-- H2 (for test) -->
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] probably, `h2` is unnecessary.

File: terasoluna-gfw-web-jsp/pom.xml [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+
+        <!-- == Begin Unit Test == -->
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.hamcrest</groupId>
+            <artifactId>hamcrest-library</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] probably, `hamcrest-library` is unnecessary.

File: terasoluna-gfw-web-jsp/pom.xml [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>joda-time</groupId>
+            <artifactId>joda-time</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] probably, `joda-time` is unnecessary.

File: terasoluna-gfw-web-jsp/pom.xml [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>terasoluna-gfw-parent</artifactId>
+        <groupId>org.terasoluna.gfw</groupId>
+        <version>5.1.0-SNAPSHOT</version>
+        <relativePath>../terasoluna-gfw-parent/pom.xml</relativePath>
+    </parent>
+    <artifactId>terasoluna-gfw-web-jsp</artifactId>
+    <packaging>jar</packaging>
+    <name>TERASOLUNA Server Framework for Java (5.x) for web</name>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web-jsp/pom.xml]: - [ ] for web -> for jsp

File: terasoluna-gfw-web/pom.xml [hunk 14]
```diff
@@ -82,18 +82,13 @@
         <!-- == Begin Tomcat == -->
         <!-- Servlet API 3.0 & JSP API 2.2 -->
         <dependency>
-            <groupId>org.apache.tomcat</groupId>
-            <artifactId>tomcat-jsp-api</artifactId>
-            <scope>provided</scope>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web/pom.xml]: - [ ] please add `provided` scope

File: terasoluna-gfw-web/pom.xml [hunk 15]
```diff
@@ -82,18 +82,13 @@
         <!-- == Begin Tomcat == -->
         <!-- Servlet API 3.0 & JSP API 2.2 -->
```
Comment (kazuki43zoo) [inline terasoluna-gfw-web/pom.xml]: - [ ] please modify comment. (Remove "JSP API 2.2")

**Comment by kazuki43zoo** [comment]
- [ ] Please remove `functions.tld` and `functions.tld` on `terasoluna-gfw-web`

**Comment by kazuki43zoo** [comment]
@btfurukawatkr @ikeyat 

I've added some comments. Please check.

**Comment by ikeyat** [comment]
@kazuki43zoo 
Thanks!

**Pull request merged**